src/output.ts

```diff
@@ -8,44 +8,73 @@ export default () => ({
   typescript: typescript(),
 });
 
+function combineFlowExactTypes(input: string[]): string {
+  if (input.length === 0) {
+    return '';
+  } else if (input.length === 1) {
+    return input[0];
+  } else {
+    return '{|' + input.map(type => `...${type}`).join(`,${EOL}`) + '|}';
+  }
+}
+
+function getNameForFallbackable(name: string): string {
+  return 'Fallbackable' + name[0].toUpperCase() + name.slice(1);
+}
+
 function flow() {
   let interfacesOutput = '';
+
+  const fallbackSet: Set<string> = new Set();
+
   for (const item of interfaces) {
     if (interfacesOutput) {
       interfacesOutput += EOL + EOL;
     }
 
-    const extendList = item.extends.map(extend => extend.name + stringifyGenerics(extend.generics, true)).join(' & ');
+    const extendList = combineFlowExactTypes(
+      item.extends.map(extend => extend.name + stringifyGenerics(extend.generics, true)),
+    );
+
     interfacesOutput += 'export type ';
     interfacesOutput += item.name + stringifyGenerics(item.generics);
     interfacesOutput += ' = ' + extendList;
 
     if (item.properties.length > 0) {
       if (extendList) {
+        // ?? Is this actually getting used?
```

Comment:
Btw, @frantic is this codepath actually gonna get hit? I wonder this is a leftover from a refactor or something

Reply:
No, it won't hit at the moment. But it wouldn't hurt if interfaces could extend other interfaces and have properties in the same time. I thought about something like this. What do you think?

```diff
@@ -14,7 +14,7 @@ function combineFlowExactTypes(input: string[]): string {
   } else if (input.length === 1) {
     return input[0];
   } else {
-    return '{|' + input.map(type => `...${type}`).join(`,${EOL}`) + '|}';
+    return input.map(type => `...${type}`).join(`,${EOL}`);
   }
 }

@@ -32,22 +32,14 @@ function flow() {
       interfacesOutput += EOL + EOL;
     }

-    const extendList = combineFlowExactTypes(
-      item.extends.map(extend => extend.name + stringifyGenerics(extend.generics, true)),
-    );
-
     interfacesOutput += 'export type ';
     interfacesOutput += item.name + stringifyGenerics(item.generics);
-    interfacesOutput += ' = ' + extendList;
+    interfacesOutput += ' = {|' + EOL;
+    interfacesOutput += combineFlowExactTypes(
+      item.extends.map(extend => extend.name + stringifyGenerics(extend.generics, true)),
+    );

     if (item.properties.length > 0) {
-      if (extendList) {
-        // ?? Is this actually getting used?
-        interfacesOutput += ' & ';
-      }
-
-      interfacesOutput += '{|' + EOL;
-
       for (const property of item.properties) {
         if (isAliasProperty(property)) {
           const generics = stringifyGenerics(property.generics, true);
@@ -73,9 +65,9 @@ function flow() {

         interfacesOutput += EOL;
       }
-
-      interfacesOutput += '|}';
     }
+
+    interfacesOutput += '|}';
```

Reply:
This seems to be more complicated than I expected. I added a TODO for now.